File: 2025/day1/sol.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"time"
)

func main() {
	start := time.Now()
	file, err := os.Open("bigboy.txt")
	if err != nil {
		fmt.Println(err)
	}
	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}
	pointer := 50
	count := 0
	for _, line := range lines {
		value, _ := strconv.Atoi(line[1:])
		if line[0] == 'L' {
			value = value * -1
		}
		pointer = (pointer + value) % 100
		if pointer == 0 {
			count++
		}
	}
	fmt.Println(count)

	pointer = 50
	count2 := 0
	for _, line := range lines {
		value, _ := strconv.Atoi(line[1:])
		sign := 1
		if line[0] == 'L' {
			sign = -1
		}
		for range value {
			pointer = (pointer + sign) % 100
			if pointer == 0 {
				count2++
			}
		}
		if pointer == 0 {
			count2++
		}
	}
	fmt.Println(count2)
	fmt.Println("Execution time:", time.Since(start).Seconds())
}

File: 2025/day1/sol.py
import time
start_time = time.time()

with open('input.txt') as file:
    lines = [line.rstrip() for line in file.readlines()]
#print(lines)
pointer = 50
count = 0
count2 = 0

for line in lines:
    if line[0]=='R':
        value = int(line[1:])
    else:
        value = int(line[1:])*-1
    pointer = (pointer+value)%100
    if(pointer==0):
        count += 1
print(count)

pointer = 50
count2 = 0
for line in lines:
    if line[0]=='R':
        sign, value = 1, int(line[1:])
    else:
        sign, value = -1, int(line[1:])
    for i in range(value):
        pointer = (pointer+sign) % 100
        if(pointer==0):
            count2 += 1
print(count2)

end_time = time.time()
execution_time = end_time - start_time
print(f"Execution time: {execution_time} seconds")